File: backend/node-services/apps/message-service/src/health.controller.ts
import { Controller, Get } from '@nestjs/common';

@Controller()
export class HealthController {
  @Get('health')
  health() {
    return {
      status: 'ok',
      service: 'message-service',
      timestamp: new Date().toISOString(),
    };
  }
}

File: backend/java-services/services/ai-service/src/main/java/iuh/cnm/vnalo/aiservice/controller/HealthController.java
package iuh.cnm.vnalo.aiservice.controller;

import iuh.cnm.vnalo.aiservice.dto.ApiResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.Instant;
import java.util.Map;

@RestController
public class HealthController {

    @GetMapping("/health")
    public ResponseEntity<ApiResponse<Map<String, Object>>> health() {
        return ResponseEntity.ok(ApiResponse.ok(Map.of(
                "service", "ai-service",
                "status", "UP",
                "timestamp", Instant.now().toString()
        )));
    }
}
